### ISSUE TITLE
The problem is Texture And the lines 

### ISSUE THREAD

**Issue by Sorshx** [godotengine/godot#86575]
### Tested versions

4.2

### System information

Mobile / Android

### Issue description

Hello, the problem is when adding Texture To a solid, for example, a large world of approx 30kilometer Width and length And raise the UV1 to 30,000 And activate Triplanar In the middle of the world there is no problem, but when you move away from the center of the world, there is a problem of lines and squares as well. These are some pictures and this project. 

In the first picture, there is no problem because it is in the center of the world 
![Screenshot_٢٠٢٣١٢٢٨-١٣٣٩٥٧](https://github.com/godotengine/godot/assets/153130532/404357a4-3062-4235-8bc5-0f38131d64d6)

The second and third when moving away from the center of the world 

![Screenshot_٢٠٢٣١٢٢٨-١٣٤٢٢٥](https://github.com/godotengine/godot/assets/153130532/a280ad3f-a237-41f1-bc55-0528660a62f0)
![Screenshot_٢٠٢٣١٢٢٨-١٣٤٢٣٣](https://github.com/godotengine/godot/assets/153130532/04c58d84-a7da-4f18-b967-6af817cbdef1)

This project 

[the problem(Copy).zip](https://github.com/godotengine/godot/files/13785803/the.problem.Copy.zip)

### Steps to reproduce

I want it to stay the same filter If the squares and lines visible on the line disappear Texture 

### Minimal reproduction project (MRP)

Thank you for helping us

**Comment by rsubtil** [comment]
Seems likely to be float precision issues. Check this page for general guidance on what to do when dealing with very large worlds: https://docs.godotengine.org/en/stable/tutorials/physics/large_world_coordinates.html. 

However, from that page:
> [Triplanar mapping](https://docs.godotengine.org/en/stable/tutorials/3d/standard_material_3d.html#doc-standard-material-3d-triplanar-mapping) doesn't benefit from increased precision. Materials using triplanar mapping will exhibit visible jittering when far away from the world origin.

**Comment by Sorshx** [comment]
Can you explain to me, I don't understand what's going on

**Comment by rsubtil** [comment]
> Can you explain to me, I don't understand what's going on

When dealing with very large values, floats start losing precision, and may no longer be able to encode a specific value without rounding it to a noticeably different value.

For example, between values `8388608` and `16777216`, the precision has degraded to only `0.5`, and you're only able to encode values on that increment, such as `8388608.0, 8388608.5, 8388609.0` and so forth. If you try to encode `8388608.3`, it gets rounded to either `8388608.0` or `8388608.5` (not sure which one right now), causing either an error of `0.3` or `0.2` regarding the original value.

Long story short; this is a limitation of how floating point numbers are encoded. These problems only become noticeable if you work with values that begin to approach these limits. You can compile the engine with double precision floats to reduce the issue, but still, your world is already too big. There are other techniques you can do to prevent this, such as [origin shifting](https://manuel-rauber.com/2022/04/06/floating-origin-in-unity/).

**Comment by Sorshx** [comment]
❤Thank you. I searched for months and finally found the solution. Thank you

**Comment by Calinou** [comment]
Closing per the above comments.
